File: server/InventoryService/inventory-service/src/main/java/com/nutrihub/inventoryservice/infrastructure/messaging/events/OrderPlacedEvent.java
package com.nutrihub.inventoryservice.infrastructure.messaging.events;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record OrderPlacedEvent(
        UUID orderId,
        UUID customerId,
        List<OrderItemEvent> items,
        double totalAmount,
        LocalDateTime placedAt
) {
    public record OrderItemEvent(
            UUID productId,
            String productName,
            int quantity,
            double unitPrice
    ) {}
}

File: server/NotificationService/NotificationService.Application/Events/OrderPlacedEvent.cs


namespace NotificationService.Application.Events
{
    public record OrderPlacedEvent(
        Guid OrderId,
        Guid CustomerId,
        List<OrderItemEvent> Items,
        double TotalAmount,
        DateTime PlacedAt
    );

    public record OrderItemEvent(
    Guid ProductId,
    string ProductName,
    int Quantity,
    double UnitPrice
);

}
